### ISSUE TITLE
Branch Workflow

### ISSUE THREAD

**Issue by elsylambert** [unfoldingWord/tc-create-app#61]
Need clear definition of how fork/branch effects the main branch

**Comment by klappy** [comment]
@jag3773 Any chance you could review the above and let me know if it's ready for Larry to review?

**Comment by jag3773** [comment]
Looks good, note this is also written up at https://forum.door43.org/t/protected-branch-workflow/76 for comparison.

If it's not *too hard*, it would be nice if tC Create could provide a link to create a pull request.  So step 7 could be a button/link in the tC Create app that takes the user directly to a URL *like this*: https://git.door43.org/jag3773/testing_/compare/master...jag3773-patch-1

Visiting that URL would allow the logged in user to start and finish the merging step without having to hunt it down in DCS.

**Comment by klappy** [comment]
I imagine that long term we will have a notification and remediation aid for step 1 as well.

We can implement it without the button(s) for step 1  and 7 at first and follow up with those buttons soon after.

**Comment by lrsallee** [comment]
Looks good to me. I also really like Jesse's suggestion above for linking directly to the PR. That's a step many will not be able to do.

**Comment by klappy** [comment]
@jag3773 We could have another button that compares your branch with master (inverse of step 7) to manually check if your branch is behind and submit a PR to update your branch with latest changes on master.

Thoughts?

**Comment by klappy** [comment]
## Issues to create for Epic:
1. Create and Implement helpers to check if Edit/Branch workflow is needed (proposed step 3).
2. Create and Implement button to create PR on DCS (proposed step 7).
3. Create and Implement button to compare branch with master (proposed step 1).

@jag3773 Can you confirm that this list appears to cover the necessary changes?
Also, for the branch workflow, what branch naming convention do we want to use?

**Comment by jag3773** [comment]
> @jag3773 We could have another button that compares your branch with master (inverse of step 7) to manually check if your branch is behind and submit a PR to update your branch with latest changes on master.
> 
> Thoughts?

That inverse function would be nice as it's something that DCS itself does not make easy.

I think the steps you've outlined are correct.

For branch naming I might use a pattern like DCS itself uses, something like:

[username]-tc-create-[#]

So my first branch would be `jag3773-tc-create-1`. The app may only support 1 branch right now but having the integer would make it easier to allow multiple branches later if the workflow supports it.

**Comment by klappy** [comment]
@birchamp Issues created, this issue converted into an Epic, and new issues associated with this Epic.

**Comment by klappy** [comment]
@birchamp Looks like I underestimated the readiness of the Gitea React Toolkit. 

The code exists in the primary functions of the core, but not accessible through the components used. I need to add another step/issue to expose the branch/ref selection through the components.

**Comment by klappy** [comment]
@elsylambert This should be available to test in `beta.9` and above here: https://develop--tc-create-app.netlify.com/

I have attempted to give you write access to the 'unfoldingWord' organization on bg server to test. This moves all editing to the bg server as well to ensure we don't mess with production data right now.

To test you must: 
1. Have access to write to the source repos (I added you).
2. Use English as your target language (This is editing the original).

The rest should happen behind the scenes. This is not intended to make any changes to any other combination of languages or permissions.
